REGRESSION ( Sonoma?): [ Sonoma ] TestWebKitAPI.WebKit2CustomProtocolsTest.LoadInvalidScheme is a consistent timeout

cdumez opened the pull request:
#### a4ce5c78105f46b76c346e3eddd7c41ec44ae96e
<pre>
REGRESSION ( Sonoma?): [ Sonoma ] TestWebKitAPI.WebKit2CustomProtocolsTest.LoadInvalidScheme is a consistent timeout
<a href="https://bugs.webkit.org/show_bug.cgi?id=263576">https://bugs.webkit.org/show_bug.cgi?id=263576</a>
rdar://117388019

Reviewed by Brent Fulgham.

The test tries calling [WKWebView loadRequest] with a URL that has an invalid scheme to make sure
no crash occurs and the load doesn&apos;t occur. We used to call the didFailProvisionalNavigation
navigation delegate. However, on Sonoma, we fail the navigation policy decision because the
request is invalid, which means that didFailProvisionalNavigation no longer gets called but
no crash or navigation occurs. Presumably, the behavior change is on CFNetwork side where they
no longer construct a valid NSURLRequest from this invalid URL. The new behavior is not incorrect
so I am simply updating the test to check that no crash or navigation occurs, instead of expecting
didFailProvisionalNavigation to get called.

* Tools/TestWebKitAPI/Tests/WebKitObjC/CustomProtocolsInvalidScheme.mm:
(-[LoadInvalidSchemeDelegate webView:didFinishNavigation:]):
(-[LoadInvalidSchemeDelegate webViewWebContentProcessDidTerminate:]):
(TestWebKitAPI::TEST):
(-[LoadInvalidSchemeDelegate webView:didFailProvisionalNavigation:withError:]): Deleted.

Canonical link: <a href="https://commits.webkit.org/269749@main">https://commits.webkit.org/269749@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/16519df413e9168bb5ac7e4512157b4c9cf26979

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/23483 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/1599 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/24606 "Built successfully") | [⏳ 🛠 wpe ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/21679 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/23752 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/3150 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/24024 "Built successfully") | [⏳ 🧪 wpe-wk2 ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/23725 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/1144 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/20319 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/26240 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/944 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/21217 "Passed tests") | [⏳ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/GTK-WK2-Tests-EWS "Waiting to run tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/21401 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/21479 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/25257 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/929 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/18648 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/898 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5598 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/1341 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/1204 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Tools/TestWebKitAPI/Tests/WebKitObjC/CustomProtocolsInvalidScheme.mm:
```diff
@@ -74,8 +80,10 @@ - (void)webView:(WKWebView *)webView decidePolicyForNavigationAction:(WKNavigati
     webView.get().navigationDelegate = loadDelegate.get();
 
     [webView loadRequest:[NSURLRequest requestWithURL:[NSURL URLWithString:@"ht'tp://www.webkit.org"]]];
+    Util::runFor(200_ms);
```
This 200_ms wait seems gross; it would be nice if we could avoid this arbitrary waiting, given how slow our tests run already.

cdumez replied:
I can reduce the timeout. Sadly, I don't believe there is currently a better way. No delegate calls happens currently, since the load doesn't even attempt navigation policy decision with the client.

brentfulgham reviewed (approved):
I think this change is fine, but it would be better if we could absolutely know if the navigation is not going to be allowed, versus only being able to confirm the navigation takes longer than 200 ms.

cdumez committed:
REGRESSION ( Sonoma?): [ Sonoma ] TestWebKitAPI.WebKit2CustomProtocolsTest.LoadInvalidScheme is a consistent timeout
https://bugs.webkit.org/show_bug.cgi?id=263576
rdar://117388019

Reviewed by Brent Fulgham.

The test tries calling [WKWebView loadRequest] with a URL that has an invalid scheme to make sure
no crash occurs and the load doesn't occur. We used to call the didFailProvisionalNavigation
navigation delegate. However, on Sonoma, we fail the navigation policy decision because the
request is invalid, which means that didFailProvisionalNavigation no longer gets called but
no crash or navigation occurs. Presumably, the behavior change is on CFNetwork side where they
no longer construct a valid NSURLRequest from this invalid URL. The new behavior is not incorrect
so I am simply updating the test to check that no crash or navigation occurs, instead of expecting
didFailProvisionalNavigation to get called.

* Tools/TestWebKitAPI/Tests/WebKitObjC/CustomProtocolsInvalidScheme.mm:
(-[LoadInvalidSchemeDelegate webView:didFinishNavigation:]):
(-[LoadInvalidSchemeDelegate webViewWebContentProcessDidTerminate:]):
(TestWebKitAPI::TEST):
(-[LoadInvalidSchemeDelegate webView:didFailProvisionalNavigation:withError:]): Deleted.

Canonical link: https://commits.webkit.org/269749@main
